services/iam/src/routes/general.js

```diff
@@ -54,6 +54,46 @@ router.post('/login', authMiddleware.authenticate, authMiddleware.accountIsEnabl
 
 });
 
+router.post('/register', async (req, res, next) => {
+    const { username ,firstname,password,lastname,companyName} = req.body;
+    try {
+        const doc = await TenantDAO.findOne({ name: companyName });
+        if(doc) {
+            return next({message: "Account already exists",status: 409})
+        } else {
+        const props = {
+            name: tenantName, 
+            status:CONSTANTS.STATUS.ACTIVE
+        }
+        const tenant = await TenantDAO.create({props});
+        const userObj = { 
+            tenant: tenant._id,
+            username,
+            firstname,
+            lastname,
+            password, 
+            permissions:['tenant.all'], 
+            status: CONSTANTS.STATUS.ACTIVE  
+        };
+        const existingUser = await AccountDAO.findOne({ username: userObj.username });
```

Comment:
What do we do if we throw an error here but the tenant was already created. Should we perform this check earlier along with the tenant check?